### DISCUSSION TITLE
Can someone share a simple example of celery + flask application? Especially that I load a model into a celery but not into flask?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [celery/celery#7770] [Q&A]
Can someone share a simple example of celery + flask application? Especially that I load a model into a celery but not into flask?

how to properly integrate flask with celery? do I need to integrate it?

I have read only some article that recommend doing something like that from flask import Flask from celery import Celery

app = Flask(name) app.config['CELERY_BROKER_URL'] = 'redis://localhost:6379/0' app.config['CELERY_RESULT_BACKEND'] = 'redis://localhost:6379/0'

celery = Celery(app.name, broker=app.config['CELERY_BROKER_URL']) celery.conf.update(app.config)

Do o need to it? Why does celery needs to know about flask app? celery = Celery(app.name, broker=app.config['CELERY_BROKER_URL']

What are the advantages of doing that? Isn't it better just to separate them?

How do i do this if in my flask application i load large model (pytorh) and i want to load it only for some workers.

**Comment by martinenkoEduard** [comment]
What is the more correct way to use celery?
This - https://github.com/matthieugouel/python-flask-celery-example
or this - https://github.com/alyssaq/celery-flask-demo?

**Reply by pfk-beta** [reply]
IMHO second example is easier to read, understand and modify.

**Comment by martinenkoEduard** [comment]
Why do we need a shared context with flask? What does it add to our application?
